Source/Shared/Backend/k8s/MicroserviceResources.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import { injectable } from 'tsyringe';
+import { KubeConfig, AppsV1Api} from '@kubernetes/client-node';
+
+import { Configuration } from '@shared/backend/configuration/Configuration';
+import { ILogger } from '@shared/backend/logging';
+import { Context } from '@shared/backend/web';
+
+import { IMicroserviceResources } from './IMicroserviceResources';
+
+@injectable()
+export class MicroserviceResources extends IMicroserviceResources {
+    private config: KubeConfig;
+    private appsV1Api: AppsV1Api;
+
+    constructor(configuration: Configuration, private logger: ILogger) {
+        super();
+        this.config = this.getKubernetesConfig(configuration);
+        this.appsV1Api = this.config.makeApiClient(AppsV1Api);
+    }
+
+    getDeployments = async (namespace: string, ctx: Context) => {
+        const { body } = await this.appsV1Api.listNamespacedDeployment(
+            namespace,
+            undefined,
+            undefined,
+            undefined,
+            undefined,
+            undefined,
+            undefined,
+            undefined,
+            undefined,
+            undefined,
+            undefined,
+            {
+                headers: {
+                    'User-ID': ctx.userId,
+                    'Tenant-ID': ctx.tenantId,
+                }
+            });
+        return body.items;
+    };
```

Comment:
We'd probably want to check the returned response in case of some failure and have different logging for them/errors to throw, something along the lines of having different errors if they come from the impersonator or kubernetes.